### PR TITLE
Adding a NO_SHOOT terrain Flag and example of use

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary

SUMMARY: Content "Terrain Flag 'NO_SHOOT'"

#### Purpose of change
Currently terrain must be hardcoded to stop bullets from passing through.  This adds a flag that when used zeroes out damage from ranged attacks.

#### Describe the solution
Adds terrain flag and effects.

#### Describe alternatives you've considered
anothersimulacrum gave me sample lines for map.cpp which meant all I had to do was add the flags.  

#### Testing
TBD

### CODE REVIEW

File: src/map.cpp [hunk 1]
```diff
@@ -3666,9 +3666,12 @@ void map::shoot( const tripoint &p, projectile &proj, const bool hit_items )
     };
 
     ter_id terrain = ter( p );
-    if( terrain == t_wall_wood_broken ||
-        terrain == t_wall_log_broken ||
-        terrain == t_door_b ) {
+    if( terrain->has_flag( "TFLAG_NO_SHOOT" ) ) {
```
Comment (Aivean) [inline src/map.cpp:3669]: Since you've added "TFLAG_NO_SHOOT" as a bitflag, use it instead:
```cpp
if( terrain->has_flag( TFLAG_NO_SHOOT ) ) { 
```

Comment (Maleclypse) [inline src/map.cpp:3669]:
```suggestion
    if( terrain->has_flag( TFLAG_NO_SHOOT ) ) { 
```

Comment (Maleclypse) [reply src/map.cpp]: Thanks!

File: src/map.cpp [hunk 2]
```diff
@@ -3666,9 +3666,12 @@ void map::shoot( const tripoint &p, projectile &proj, const bool hit_items )
     };
 
     ter_id terrain = ter( p );
-    if( terrain == t_wall_wood_broken ||
-        terrain == t_wall_log_broken ||
-        terrain == t_door_b ) {
+    if( terrain->has_flag( TFLAG_NO_SHOOT ) ) {
```
Comment (Maleclypse) [inline src/map.cpp:3669]:
```suggestion
    if( terrain->has_flag( TFLAG_NO_SHOOT ) ) {
```

**Review by Aivean — APPROVED** [review]

**Pull request merged**